core/src/main/scala/kafka/network/SocketServer.scala

```
@@ -215,14 +269,19 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
     this.synchronized {
       if (!stoppedProcessingRequests)
         stopProcessingRequests()
-      requestChannel.shutdown()
+      dataPlaneRequestChannel.shutdown()
+      controlPlaneRequestChannelOpt.map(_.shutdown())
     }
     info("Shutdown completed")
   }
 
   def boundPort(listenerName: ListenerName): Int = {
     try {
-      acceptors.get(endpoints(listenerName)).serverChannel.socket.getLocalPort
+      if (dataPlaneAcceptors.containsKey(endpoints(listenerName))) {
+        dataPlaneAcceptors.get(endpoints(listenerName)).serverChannel.socket.getLocalPort
+      } else {
+        controlPlaneAcceptorOpt.map (_.serverChannel.socket().getLocalPort).head
```

head on an Option doesn't guarantee an actual value. So, we need to throw an exception if Option is empty.

---

Agreed.